### PR TITLE
Consolidates auth check into one place

### PR DESCRIPTION
**Pull request by dagi**
Consolidates the auth check into one place. This change ensures all endpoints require an authenticated user.

### CODE REVIEW

File: src/btwa_api/main.py [hunk 1]
```diff
@@ -48,6 +54,12 @@
     return response
 
 
+@app.middleware("http")
+async def auth_handler(request: Request, call_next, 
+                       credentials: HTTPBasicCredentials = Depends(security), db: Database = Depends(get_db)):
+    check_user_authenticated(db, credentials)
```
Comment (petr-k) [inline src/btwa_api/main.py:60]: Given this is an `async` fn, shouldn't the `check_user_authenticated` (which contains blocking IO code) call here be offloaded to a thread pool? What do you think @Kobzol ?

Comment (Kobzol) [reply src/btwa_api/main.py:60]: Good point. It should be enough to make the middleware function sync.

Comment (petr-k) [reply src/btwa_api/main.py:60]: So for sync functions, FastAPI will run them on a threadpool by itself, right?

Comment (Kobzol) [reply src/btwa_api/main.py:60]: Yes.

**Pull request closed without merging**